### PR TITLE
Implemented selectors, UI change and tests for permission-based report availablity

### PR DESCRIPTION
**Pull request by ZacharyKlein**
@mkimberlin Not sure if I have the correct permissions for all of these reports, I just made my best guess - please review and advise.

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/security/permissions/Permissions.java [hunk 1]
```diff
@@ -15,6 +15,7 @@ public enum Permissions {
     CAN_VIEW_ANNIVERSARY_REPORT,
     CAN_VIEW_BIRTHDAY_REPORT,
     CAN_VIEW_PROFILE_REPORT,
+    CAN_VIEW_CHECKINS_REPORT,
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/security/permissions/Permissions.java:18]: One more small thing needed with this. We need to load this permission into the DB and assign it to the ADMIN role in the load test data sql file.

**Review by mkimberlin — DISMISSED** [review]

File: web-ui/src/context/selectors.test.js [hunk 2]
```diff
@@ -1195,5 +1195,99 @@ describe("Selectors", () => {
...
+
+
+        const testState1 = {
+            userProfile: {
+                firstName: "Big",
+                lastName: "Boss",
+                role: "ADMIN",
+                permissions: [{permission: "CAN_VIEW_SKILLS_REPORT"}, {permission: "CAN_VIEW_ANNIVERSARY_REPORT"}],
+            }
+        };
```
Comment (pieperm) [inline web-ui/src/context/selectors.test.js:1279]: This variable is not being used in this test

File: web-ui/src/context/selectors.test.js [hunk 3]
```diff
@@ -1195,5 +1195,99 @@ describe("Selectors", () => {
...
```
Comment (pieperm) [inline web-ui/src/context/selectors.test.js:1256]: This variable is not being used in this test

File: web-ui/src/context/selectors.test.js [hunk 4]
```diff
@@ -1195,5 +1195,99 @@ describe("Selectors", () => {
...
+
+
+        const testState1 = {
+            userProfile: {
+                firstName: "Big",
+                lastName: "Boss",
+                role: "ADMIN",
+                permissions: [{permission: "CAN_VIEW_CHECKINS"}, {permission: "CAN_VIEW_ANNIVERSARY_REPORT"}],
+            }
+        };
```
Comment (pieperm) [inline web-ui/src/context/selectors.test.js:1233]: This variable is not being used in this test

**Review by pieperm — COMMENTED** [review]
Most of Check-Ins uses two spaces for indents in JavaScript. We could change it to be four spaces if we want, but maybe it should be a separate issue. The diff on this PR makes it harder to tell which changes are part of the feature and which changes are due to the auto-formatting.

**Review by mkimberlin — APPROVED** [review]

**Review by pieperm — DISMISSED** [review]
There seems to be something broken with `getReportLinks` in Menu.jsx. Despite having logging in as SUPER and having all the necessary permissions for viewing reports, none of those pages appear in the menu. I can still access the report pages by manually entering the URL. I think the issue may be caused by the selectors for `selectHasBirthdayAnniversaryReportPermission`, `selectHasCheckinsReportPermission`, etc.

A snippet of the list of the user's permissions from the console:

```
[
  ...
  "CAN_VIEW_SKILLS_REPORT",
  "CAN_VIEW_RETENTION_REPORT",
  "CAN_VIEW_ANNIVERSARY_REPORT",
  "CAN_VIEW_BIRTHDAY_REPORT",
  "CAN_VIEW_PROFILE_REPORT",
  ...
]
```

**Review by pieperm — APPROVED** [review]

**Review by mkimberlin — APPROVED** [review]

**Pull request merged**